owlplug-client/src/main/java/com/owlplug/store/controllers/StoreController.java

```diff
@@ -344,14 +344,25 @@ public boolean installBundle(ProductBundle bundle) {
         bundle.getProduct().getName(), 
         bundle.getName());
 
+    
     String baseDirectoryPath = storeService.getBundleInstallFolder(bundle);
     String relativeDirectoryPath  = this.getPreferences().get(ApplicationDefaults.STORE_DIRECTORY_KEY, "");
 
+    Boolean shouldGroupByCreator = this.getPreferences().getBoolean(ApplicationDefaults.STORE_BY_MAKER_ENABLED_KEY, false);
+
+    //if the enduser wishes to group plugins by their creator,
+    //then we need to include the subdirectory as well.
+    if(shouldGroupByCreator) {
```

Comment:
If you've decided to group by the creator, here's where it happens. otherwise, the behavior is identical to beforehand. The only change is in the output dir.

Reply:
LGTM